Build error with clang 8 (Ubuntu cosmic)

After upgrading from clang 7 to clang 8 (as part of upgrading to the newly-released Ubuntu disco), building corefx fails with the following error:

```
  Scanning dependencies of target System.Native
  [ 38%] Building C object System.Native/CMakeFiles/System.Native.dir/pal_random.c.o
/home/roji/projects/corefx/src/Native/Unix/System.Native/pal_random.c(54,68): error G080D97BF: implicit use of sequentially-consistent atomic may incur stronger memory barriers than necessary [-Werror,-Watomic-implicit-seq-cst] [/home/roji/projects/corefx/src/Native/build-native.proj]
  Failed to build corefx native components.
                  if (!__sync_bool_compare_and_swap(&rand_des, -1, fd))
                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~                  ^
  1 error generated.
  make[2]: *** [System.Native/CMakeFiles/System.Native.dir/build.make:193: System.Native/CMakeFiles/System.Native.dir/pal_random.c.o] Error 1
  make[1]: *** [CMakeFiles/Makefile2:281: System.Native/CMakeFiles/System.Native.dir/all] Error 2
  make: *** [Makefile:130: all] Error 2
/home/roji/projects/corefx/src/Native/build-native.proj(36,5): error MSB3073: The command ""/home/roji/projects/corefx/src/Native/build-native.sh" x64 Debug Linux outconfig netcoreapp-Linux-Debug-x64 -portable" exited with code 1.

Build FAILED.

/home/roji/projects/corefx/src/Native/Unix/System.Native/pal_random.c(54,68): error G080D97BF: implicit use of sequentially-consistent atomic may incur stronger memory barriers than necessary [-Werror,-Watomic-implicit-seq-cst] [/home/roji/projects/corefx/src/Native/build-native.proj]
/home/roji/projects/corefx/src/Native/build-native.proj(36,5): error MSB3073: The command ""/home/roji/projects/corefx/src/Native/build-native.sh" x64 Debug Linux outconfig netcoreapp-Linux-Debug-x64 -portable" exited with code 1.
    0 Warning(s)
    2 Error(s)
```

--- a/src/Native/Unix/System.Native/pal_random.c
+++ b/src/Native/Unix/System.Native/pal_random.c
@@ -51,7 +51,8 @@ void SystemNative_GetNonCryptographicallySecureRandomBytes(uint8_t* buffer, int3
 
             if (fd != -1)
             {
-                if (!__sync_bool_compare_and_swap(&rand_des, -1, fd))
+                int expected = -1;
+                if (!__atomic_compare_exchange_n(&rand_des, &expected, fd, false, __ATOMIC_SEQ_CST, __ATOMIC_SEQ_CST))
                 {
                     // Another thread has already set the rand_des
                     close(fd);

